pages/transactions/[id].js

```diff
@@ -4,17 +4,59 @@ import { useTransactionsContext } from "@/utils/TransactionsContext/Transactions
 import Link from "next/link";
 import { ChevronLeft } from "lucide-react";
 import dayjs from "dayjs";
+import { Trash2 } from "lucide-react";
+import { useModalContext } from "@/utils/ModalContext/ModalContext";
+import { useState } from "react";
 
 export default function TransactionDetails() {
   const router = useRouter();
   const { id } = router.query;
 
   const { data, isLoading } = useTransactionsContext();
+  const { openModal } = useModalContext();
+
+  const [wasDeleted, setWasDeleted] = useState(null);
 
   const currentTransaction = data.find((transaction) => transaction.id === id);
 
+  if (wasDeleted)
+    return (
+      <TransactionDetailsWrapper>
+        <TransactionDetailsHeader>
+          <IconLink href={"/"}>
+            <ChevronLeft />
+            Zurück
+          </IconLink>
+          <TransactionDetailsHeaderHeading>
+            Details
+          </TransactionDetailsHeaderHeading>
+        </TransactionDetailsHeader>
```

Comment:
DRY: Here is so much repetition in the component. If you change anything in the container you need to change it three times the same. 
Better: Return the container and only conditionally the part that changes.